src/Raven.Client/Json/Serialization/NewtonsoftJson/DefaultRavenContractResolver.cs

```diff
@@ -193,7 +193,7 @@ private bool MembersToFilterOut(MemberInfo info)
             var fieldInfo = info as FieldInfo;
             if (fieldInfo != null)
             {
-                if (fieldInfo.IsPublic == false)
+                if ((fieldInfo.IsPublic == false) && (fieldInfo.IsDefined(typeof(JsonDeserializationDoNotIgnoreAttribute)) == false))
```

Comment:
What about property? Handled 20 lines below.

Reply:
no need. we are not filtering out non-public properties